src/os/bluestore/BlueFS.cc

```diff
@@ -812,6 +812,7 @@ void BlueFS::_init_alloc()
         name += devnames[id];
       else
         name += to_string(uintptr_t(this));
+      block_reserved[id] = p2roundup(block_reserved[id], alloc_size[id]);
```

Comment:
What would happen if there is an OSD which has got existing allocations within that "increased" reserved extent? init_add_free() call below wouldn't mark this space as free any more and hence subsequent init_rm_free() call hitting this space might fail. 
Just speculating so far though..

Reply:
Well, here is a real log output from an osd without your patch:
2024-10-31T12:17:25.415+0300 7fb1d16ec740 10 AvlAllocator init_add_free 0x1000~27ffff000
2024-10-31T12:17:25.415+0300 7fb1d16ec740  1 bluefs _init_alloc new, id 1, allocator name bluefs-db, allocator type hybrid, capacity 0x10000000
00, reserved 0x2000, block size 0x100000, max alloc size 0x100000
2024-10-31T12:17:25.415+0300 7fb1d16ec740 10 AvlAllocator AvlAllocator 0x1000000000/100000
2024-10-31T12:17:25.415+0300 7fb1d16ec740 10 AvlAllocator init_add_free 0x2000~fffffe000
2024-10-31T12:17:25.415+0300 7fb1d16ec740  1 bluefs _init_alloc shared, id 2, capacity 0x4b00000000, block size 0x10000
...
2024-10-31T12:17:25.468+0300 7fb1d16ec740 10 hybrid::init_rm_free offset 0x2000 length 0x100000

So as you can see in the last line there is existing extent 0x2000~100000 which might cause the issue...

Reply:
Legacy free list managers' (both bitmap and null) behavior is an open question with this patch as well IMO.